MaterialDesignThemes.Wpf/PaletteHelper.cs

```diff
@@ -115,6 +116,33 @@ public void ReplaceAccentColor(string name)
 
             ReplaceAccentColor(swatch);
         }
+        #region Pinvoke
+            public struct PowerState
+            {
+                public ACLineStatus ACLineStatus;
+                public byte BatteryFlag;
+                public Byte BatteryLifePercent;
+                public Byte Reserved1;
+                public Int32 BatteryLifeTime;
+                public Int32 BatteryFullLifeTime;
+            }
+            [DllImport("Kernel32", EntryPoint = "GetSystemPowerStatus")]
+            private static extern bool GetSystemPowerStatusRef(PowerState sps);
+            
+        #endregion
+        public static PowerState GetPowerState()
+        {
+            PowerState state = new PowerState();
+            if (GetSystemPowerStatusRef(state))
+                return state;
+
+            throw new ApplicationException("Unable to get power state");
+        }
+        // Note: Underlying type of byte to match Win32 header
+        public enum ACLineStatus : byte
+        {
+            Offline = 0, Online = 1, Unknown = 255
+        }
```

Comment:
I'm not keen how this is shoe-horned in here.  It makes no consideration of the the rest of the library.

* Is this really an issue?
* Should we be _disabling_ all animations on battery?
  * This is a signification piece of work
  * It should be configurable
  * Maybe if this is a real requirement we should log an issue so I can come up with a more encompassing solution.